types/bitcoin.types.ts

```diff
@@ -107,3 +110,40 @@ export const callOptionsSchema = z
     message: typesAndDataExclusivityRefineRule.message,
     path: typesAndDataExclusivityRefineRule.path,
   });
+
+export const memoDepositAndCallOptionsSchema = z.object({
+  amount: validAmountSchema,
+  bitcoinApi: z.string().url(),
+  data: hexStringSchema.optional(),
+  gasPriceApi: z.string().url(),
+  gateway: z.string(),
+  name: z.string().optional().default(DEFAULT_ACCOUNT_NAME),
+  networkFee: z.string(),
+  privateKey: z.string().optional(),
+  receiver: z.string().optional(),
+});
+
+export const memoDepositOptionsSchema = z.object({
+  amount: z.string().refine((val) => !isNaN(Number(val)) && Number(val) > 0, {
+    message: "Amount must be a valid positive number",
+  }),
+  bitcoinApi: z.string().url(),
+  data: hexStringSchema.optional(),
+  gasPriceApi: z.string().url(),
+  gateway: z.string(),
+  name: z.string().optional().default(DEFAULT_ACCOUNT_NAME),
+  networkFee: z.string(),
+  privateKey: z.string().optional(),
+  receiver: z.string().optional(),
+});
+
+export const memoCallOptionsSchema = z.object({
+  bitcoinApi: z.string().url(),
+  data: hexStringSchema.optional(),
+  gasPriceApi: z.string().url(),
+  gateway: z.string(),
+  name: z.string().optional().default(DEFAULT_ACCOUNT_NAME),
+  networkFee: z.string(),
+  privateKey: z.string().optional(),
+  receiver: z.string().optional(),
+});
```

Comment:
It seems like these memo schemas don't have the same validation rules as the inscription schemas (i.e.: name, pk exclusivity)

Reply:
We do have validation using commander's syntax:

```ts
export const addCommonOptions = (command: Command) => {
  return command
    .addOption(
      new Option("--private-key <key>", "Bitcoin private key").conflicts([
        "name",
      ])
    )
    .addOption(
      new Option("--name <name>", "Account name")
        .default(DEFAULT_ACCOUNT_NAME)
        .conflicts(["private-key"])
    );
};
```

I think we should decide which validation system we prefer for commands: commander or zod. Duplicating the logic in two places is not ideal. I propose we use commander for all validation that it can handle (conflicts, choices, default values) and use zod for more complex logic.